### DISCUSSION TITLE
Spill frequency settings

### DISCUSSION THREAD

**Discussion by AskAlexSharov** [erthink/libmdbx#349] [Q&A] [answered]
- Lmdb default durtylist limit is 2^17, but mdbx’s is 2^16 
Is it ok? 

https://git.openldap.org/openldap/openldap/-/blob/master/libraries/liblmdb/midl.h#L61

https://git.openldap.org/openldap/openldap/-/blob/master/libraries/liblmdb/mdb.c#L2051

- one more question: does APPEND has any difference from UPSERT from dirty pages and spill point of view? Or no special cases here?

- Please educate me about “spill”: if pages were not modified after spill, then there will no need to to read/write them at commit time, right? (ah, no, look's like i'm crazy - `mdbx_page_flush` - does flush to disk, but it doesn't mean that pages are gone from RAM - if transaction will try to update this pages again it may find them in PageCache, right?)

**Comment by erthink** [comment, marked as answer]
Please take look to the [`MDBX_opt_txn_dp_limit`](https://libmdbx.dqdkfa.ru/group__c__settings.html#gga671855605968c3b1f89a72e2d7b71eb3a262cec5ffe0762dd2254fe6dd7b2e6c7), [`MDBX_opt_spill_max_denominator`](https://libmdbx.dqdkfa.ru/group__c__settings.html#gga671855605968c3b1f89a72e2d7b71eb3ab1150ff5efb3992daf6626909d17a1e2) and [`MDBX_opt_spill_min_denominator`](https://libmdbx.dqdkfa.ru/group__c__settings.html#gga671855605968c3b1f89a72e2d7b71eb3a5b1a57f0af52b62ac57d594f939e06a9). This definitely gives you more control in comparison to LMDB.

I.e. you can set any spilling threshold(s) that will reasonable for your cases.

**Comment by AskAlexSharov** [comment]
yes, definitely.

**Comment by erthink** [comment]
> does APPEND has any difference from UPSERT from dirty pages and spill point of view? Or no special cases here?

No special cases.

With the APPEND when the page is filled in, a new one will be added instead of splitting it. Thus you get approximately a half of dirty pages in a large appending case.
